packages/ui/src/council/hooks/useMyPastVotes.ts

```diff
@@ -1,40 +1,86 @@
+import { useMemo } from 'react'
+
 import { useMyAccounts } from '@/accounts/hooks/useMyAccounts'
 import { CastVoteOrderByInput } from '@/common/api/queries'
-import { usePagination } from '@/common/hooks/usePagination'
-import { SortOrder, toQueryOrderByInput } from '@/common/hooks/useSort'
+import { SortOrder } from '@/common/hooks/useSort'
+
+import { useGetCouncilVotesQuery } from '../queries'
+import { asVote, Vote } from '../types/Vote'
 
-import { useGetCouncilVotesCountQuery, useGetCouncilVotesQuery } from '../queries'
-import { asVote } from '../types/Vote'
+const VOTES_PER_ACCOUNT_FETCH_LIMIT = 25
+const MIN_FETCH_LIMIT = 100
 
 interface UseMyPastVotesProps {
   order: SortOrder<CastVoteOrderByInput>
-  perPage?: number
 }
 
-export const useMyPastVotes = ({ order, perPage = 5 }: UseMyPastVotesProps) => {
+export const useMyPastVotes = ({ order }: UseMyPastVotesProps) => {
   const { allAccounts } = useMyAccounts()
 
-  const where = {
-    castBy_in: allAccounts.map((account) => account.address),
-    electionRound: { isFinished_eq: true },
-  }
+  const addresses = useMemo(() => allAccounts.map((account) => account.address), [allAccounts])
 
-  const { data: countData, loading: countLoading } = useGetCouncilVotesCountQuery({ variables: { where } })
-  const totalCount = countData?.castVotesConnection.totalCount
-  const { offset, pagination } = usePagination(perPage, totalCount ?? 0, [order, totalCount])
+  const variables = useMemo(() => {
+    if (!addresses.length) return undefined
 
-  const variables = {
-    where,
-    orderBy: toQueryOrderByInput<CastVoteOrderByInput>(order),
-    limit: perPage,
-    offset,
-  }
+    return {
+      where: {
+        castBy_in: addresses,
+        electionRound: { isFinished_eq: true },
+      },
+      orderBy: [CastVoteOrderByInput.CreatedAtDesc],
+      limit: Math.max(addresses.length * VOTES_PER_ACCOUNT_FETCH_LIMIT, MIN_FETCH_LIMIT),
+    }
+  }, [addresses])
+
+  const { data, loading } = useGetCouncilVotesQuery({
+    variables,
+    skip: !variables,
+  })
+
+  const votes = useMemo(() => {
+    if (!data?.castVotes) return []
+
+    const seenAccounts = new Set<string>()
+    const latestVotes = data.castVotes.filter((vote) => {
+      if (seenAccounts.has(vote.castBy)) return false
+      seenAccounts.add(vote.castBy)
+      return true
+    })
 
-  const { data, loading } = useGetCouncilVotesQuery({ variables })
+    const parsedVotes = latestVotes.map(asVote)
+
+    return sortVotes(parsedVotes, order)
+  }, [data?.castVotes, order])
 
   return {
-    votes: data?.castVotes.map(asVote),
-    isLoading: loading || countLoading,
-    pagination: pagination,
+    votes,
+    isLoading: loading,
+  }
+}
+
+const sortVotes = (votes: Vote[], order: SortOrder<CastVoteOrderByInput>) => {
```

Comment:
This can be moved to [common/hooks/useSort.ts](https://github.com/Joystream/pioneer/tree/dev/packages/ui/src/common/hooks/useSort.ts).